File: backend/services/gallery_service.py
"""
Gallery module service
Business logic for Gallery module
"""
from datetime import datetime
from typing import List
from ..models.gallery_models import GalleryItem, CreateGalleryItemRequest, GalleryItemsResponse
import uuid


class GalleryService:
    """Service for Gallery module"""

    def __init__(self):
        # In-memory storage (stub)
        self.items: List[GalleryItem] = []

    def get_items(self) -> GalleryItemsResponse:
        """Get all gallery items"""
        return GalleryItemsResponse(
            items=self.items,
            count=len(self.items)
        )

    def create_item(self, request: CreateGalleryItemRequest) -> GalleryItem:
        """Create a new gallery item"""
        now = datetime.utcnow().isoformat()
        item = GalleryItem(
            id=str(uuid.uuid4()),
            title=request.title,
            description=request.description,
            type=request.type,
            url=request.url,
            thumbnail_url=None,
            tags=request.tags,
            created_at=now,
            updated_at=now,
            metadata={}
        )
        self.items.append(item)
        return item

    def get_item(self, item_id: str) -> GalleryItem:
        """Get item by ID"""
        for item in self.items:
            if item.id == item_id:
                return item
        raise ValueError(f"Item {item_id} not found")

    def delete_item(self, item_id: str) -> None:
        """Delete an item"""
        self.items = [i for i in self.items if i.id != item_id]


# Singleton instance
gallery_service = GalleryService()

File: frontend/src/services/modules/galleryService.ts
/**
 * Gallery Module API Service
 * Handles all API calls for Gallery (Media) module
 */

import { apiClient } from '../api';

export interface GalleryItem {
  id: number;
  title: string;
  description: string;
  media_type?: string;
  url?: string;
  thumbnail_url?: string;
  metadata?: Record<string, any>;
  tags?: string[];
  time?: string;
  module?: string;
  source_trace?: string;
  canon_bundle_id?: string;
}

export interface GalleryItemCreate {
  title: string;
  description: string;
  media_type?: string;
  url?: string;
  thumbnail_url?: string;
  metadata?: Record<string, any>;
  tags?: string[];
}

export interface GalleryItemUpdate {
  title?: string;
  description?: string;
  media_type?: string;
  url?: string;
  thumbnail_url?: string;
  metadata?: Record<string, any>;
  tags?: string[];
}

const galleryService = {
  /**
   * Get Gallery module status
   */
  async getStatus() {
    const response = await apiClient.get('/api/v1/gallery/');
    return response.data;
  },

  /**
   * Create a new gallery item
   */
  async createItem(item: GalleryItemCreate): Promise<GalleryItem> {
    const response = await apiClient.post('/api/v1/gallery/items', item);
    return response.data;
  },

  /**
   * Get all items with pagination
   */
  async getItems(params?: {
    skip?: number;
    limit?: number;
    media_type?: string;
  }): Promise<GalleryItem[]> {
    const response = await apiClient.get('/api/v1/gallery/items', { params });
    return response.data;
  },

  /**
   * Get item by ID
   */
  async getItem(id: number): Promise<GalleryItem> {
    const response = await apiClient.get(`/api/v1/gallery/items/${id}`);
    return response.data;
  },

  /**
   * Update item
   */
  async updateItem(id: number, updates: GalleryItemUpdate): Promise<GalleryItem> {
    const response = await apiClient.put(`/api/v1/gallery/items/${id}`, updates);
    return response.data;
  },

  /**
   * Delete item
   */
  async deleteItem(id: number): Promise<void> {
    await apiClient.delete(`/api/v1/gallery/items/${id}`);
  },

  /**
   * Get items by media type
   */
  async getItemsByType(mediaType: string): Promise<GalleryItem[]> {
    return this.getItems({ media_type: mediaType });
  },

  /**
   * Get recent items
   */
  async getRecentItems(limit: number = 20): Promise<GalleryItem[]> {
    return this.getItems({ limit });
  },
};

export default galleryService;
